ambari-server/src/main/java/org/apache/ambari/server/controller/KerberosHelperImpl.java

```diff
@@ -1614,14 +1614,14 @@ public int addIdentities(KerberosIdentityDataFileWriter kerberosIdentityDataFile
               // validating owner and group
               boolean differentOwners = false;
               String warnTemplate = "Keytab '{}' on host '{}' has different {}, originally set to '{}' and '{}:{}' has '{}', using '{}'";
-              if (!resolvedKeytab.getOwnerName().equals(sameKeytab.getOwnerName())) {
+              if (!StringUtils.equals(resolvedKeytab.getOwnerName(), sameKeytab.getOwnerName())) {
```

Comment:
Is it possible for `sameKeytab` to be `null`?  Do we want to protect against that?

Reply:
There is a check above

```java
if (resolvedKeytabs.containsKey(keytabFilePath)) {
    ResolvedKerberosKeytab sameKeytab = resolvedKeytabs.get(keytabFilePath);
```

Reply:
oops..  I missed that. Thanks.